lib/net/ssh/authentication/agent.rb

```diff
@@ -71,10 +71,10 @@ def connect!(agent_socket_factory = nil)
       @socket =
         if agent_socket_factory
           agent_socket_factory.call
-        elsif ENV['SSH_AUTH_SOCK'] && defined?(unix_socket_class)
```

Comment:
Your proposed fix works like a charm. Should I update the PR, or do you want to fix it by your own? My ruby skills are not good enough to come up with a test case for this.